DTS-STN/Alpha-Site
# Feature/122 failure signup form

# Description

https://trello.com/c/stGZUuwk/122-122-success-failure-page-sign-up-form

Hey all. I have completed the functionality of the signup page! There are things I did not have time for including:

1. `writing end-to-end tests`: So sorry about this but I made sure that there is unit test coverage. I have also extensively tested it manually and everything is working perfectly together. There may be a few bugs that I haven't caught but I'm thinking this can be done as a separate task and the person writing the end to end tests can fix any bugs they find through the automated tests  
2. `creating translation keys in the i18n JSON files`: This is super tedious and I don't have time unfortunately to do this. What I have made sure however is all the strings ( including the dynamic ones like the error messages ) are predictable and able to be incorporated into the i18n translations. 
3. `creating the thank you page`: Again, this was tedious and since this PR is giant enough, it best be done separately. The functionality to redirect to a `/thankyou` page is present however once the form successfully submits 

I thought it would be better to open a PR and these items can be done as separate tasks considering this PR is giant anyways and at the same time, you can get the form in to show this sprint.


## Test Instructions

1. Follow readme instructions to start mongo database and hook up notify 
2. Test the form without filling any fields, note error messages and you should see an error box with links you can click on which will take you to the error ![Screen Shot 2021-06-07 at 11 59 43 PM](https://user-images.githubusercontent.com/26819992/121120865-725f7580-c7ec-11eb-8883-0ad0f46b839f.png)
3. Fill out the form and head to http://localhost:8081 and check the collection to ensure that it is created ![Screen Shot 2021-06-08 at 12 01 38 AM](https://user-images.githubusercontent.com/26819992/121120980-b5b9e400-c7ec-11eb-8ae9-f462b1732b30.png)
4. Check your email to see if you got the validation email form to notify
5. Submit the form again with the same information and not this error at the top ![Screen Shot 2021-06-08 at 12 03 58 AM](https://user-images.githubusercontent.com/26819992/121121131-07fb0500-c7ed-11eb-90d0-5a4d0cab9ad9.png)


## Checklist


## Product and Sprint Backlog

[Trello](https://trello.com/b/ZqWtJSyh/alpha-site-board)
